### PR TITLE
[NCL-3511]: Create a scheduler for regular cleanup of temporary builds

### PR DESCRIPTION
**Pull request by jbartece**
NOT READY TO BE MERGED - missing binding to the deletion of temporary builds, more tests should be written

### CODE REVIEW

File: common/src/main/java/org/jboss/pnc/common/util/TimeUtils.java [hunk 1]
```diff
@@ -113,4 +113,15 @@ public static String generateTimestamp(boolean generationEnabled, Date dateInsta
...
+    /**
+     * Get date object representing timestamp X days ago specified by parameter
+     *
+     * @param numberOfDays Number of days to shift the time back
+     * @return New Date object representing a timestamp X days before now
+     */
+    public static Date getDateXDaysAgo(int numberOfDays) {
+        long unixMilisecondsExpirationThreshold = new Date().getTime() - (1000 * 60 * 60 * 24 * numberOfDays);
+        return new Date(unixMilisecondsExpirationThreshold);
+    }
```
Comment (matejonnet) [inline common/src/main/java/org/jboss/pnc/common/util/TimeUtils.java]: Any reason to not use a LocalDate ?
LocalDate specialOfferExpiryDate = today.plusWeeks(2).with(next(FRIDAY));
https://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html

Comment (jbartece) [reply common/src/main/java/org/jboss/pnc/common/util/TimeUtils.java]: We cannot use LocalDate as we need a "Date" object to use it for database queries.

**Comment by jbartece** [comment]
@matejonnet Ready for review

**Comment by jbartece** [comment]
retest this please

**Comment by matedo1** [comment]
retest this please

**Pull request merged**